### PR TITLE
update GTDB-tk to 2.3.2

### PR DESCRIPTION
**Pull request by maxibor**
Update to the latest GTDB-tk version.
Notable changes: 
- output files are in subdirectories (`identify`, `classify`) are are now `mv` to the current working directory.
- The latest version can make use of mash database to speedup ANI placement. See here: https://ecogenomics.github.io/GTDBTk/commands/classify_wf.html#mutually%20exclusive%20required%20arguments_repeat1
- 
 ## PR checklist

- [x] This comment contains a description of changes (with reason).
- [x] Emit the `versions.yml` file.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [x] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

File: modules/nf-core/gtdbtk/classifywf/meta.yml [hunk 1]
```diff
@@ -31,6 +31,10 @@ input:
       type: file
       description: The local copy of the taxonomic database used by GTDB-tk (unzipped copy)
       pattern: "*"
+  - mash_db:
+      type: file
+      description: The local copy of the Mash sketch database used by GTDB-tk (optional)
```
Comment (jfy133) [inline modules/nf-core/gtdbtk/classifywf/meta.yml:36]: For which contexts?

File: tests/modules/nf-core/gtdbtk/classifywf/main.nf [hunk 2]
```diff
@@ -6,18 +6,29 @@ include { GTDBTK_CLASSIFYWF } from '../../../../../modules/nf-core/gtdbtk/classi
...
+    stub:
+    """
+    touch db.msh
     """
 }
 
 workflow test_gtdbtk_classifywf {
 
     STUB_GTDBTK_DATABASE()
+    STUB_mash_database()
```
Comment (jfy133) [inline tests/modules/nf-core/gtdbtk/classifywf/main.nf:31]:
```suggestion
    STUB_MASH_DATABASE()
```

**Review by jfy133 — APPROVED** [review]
LGTM, a few minor things here and there

File: modules/nf-core/gtdbtk/classifywf/meta.yml (hunk 1 above)
Comment (jfy133) [reply modules/nf-core/gtdbtk/classifywf/meta.yml:36]: As in, why is it optional - in which cases would you want to supply such a database

Comment (maxibor) [reply modules/nf-core/gtdbtk/classifywf/meta.yml:36]: https://ecogenomics.github.io/GTDBTk/commands/classify_wf.html#mutually%20exclusive%20required%20arguments_repeat1

Comment (jfy133) [inline modules/nf-core/gtdbtk/classifywf/meta.yml:36]:
```suggestion
      description: The local copy of the Mash sketch database used by GTDB-tk if `ani_screen` mode is used (optional)
```

File: tests/modules/nf-core/gtdbtk/classifywf/main.nf [hunk 3]
```diff
@@ -28,5 +39,5 @@ workflow test_gtdbtk_classifywf {
         ]
     ]
 	
-    GTDBTK_CLASSIFYWF ( input, STUB_GTDBTK_DATABASE.out.database )
+    GTDBTK_CLASSIFYWF ( input, STUB_GTDBTK_DATABASE.out.database, STUB_mash_database.out.mash_db )
```
Comment (jfy133) [inline tests/modules/nf-core/gtdbtk/classifywf/main.nf:42]:
```suggestion
    GTDBTK_CLASSIFYWF ( input, STUB_GTDBTK_DATABASE.out.database, STUB_MASH_DATABASE.out.mash_db )
```

**Review by jfy133 — APPROVED** [review]
Assuming tests pass after my suggestions below, and has been tested outside with real data!

**Pull request merged**